scalyr_agent/json_lib/parser.py

```diff
@@ -233,20 +229,22 @@ def parse_value(self):
             elif c == "n":
                 self.__match("null", "unknown identifier")
                 return None
+            elif c is None:
```

Comment:
Add a comment to indicate what `c` being `None` represents.  Does it mean we are at the end of the stream?  Or an invalid character?

Reply:
Done. Yes, it means the end of the stream or an empty stream. Invalid character is handled by the last else clause.